File: src/utils/http/retroarch.ts
import {RetroarchRunner} from '@/models/enums/retroarch-runner'
import {TauriCommandKey, TauriCommandPayload, tauriInvoke} from '.'
import {RetroarchCore} from '@/models/enums/retroarch-core'

export const playRetroarch = async (params: {
	core: RetroarchCore
	runner: RetroarchRunner
	romPath: string
}): Promise<TauriCommandPayload<null>> => {
	return tauriInvoke(TauriCommandKey.PLAY_RETROARCH_GAME, params)
}

File: src-tauri/src/commands/retroarch.rs
use tauri::AppHandle;

use crate::{
    enums::error::Error,
    services::retroarch::{RetroarchCore, RetroarchService, RetroarchRunner},
};

#[tauri::command]
pub async fn command_play_retroarch_game(
    app_handle: AppHandle,
    runner: RetroarchRunner,
    core: RetroarchCore,
    rom_path: String,
    rom_id: i32
) -> Result<(), Error> {
    let config = RetroarchService::new(runner, core, rom_path, rom_id);
    config.play(&app_handle).await
}
